state/block.go

```diff
@@ -24,15 +25,28 @@ func NewBlockState() *blockState {
 
 var (
 	// Data prefixes
-	headerPrefix    = []byte("hdr") // headerPrefix + hash -> header
-	blockDataPrefix = []byte("hsh") // blockDataPrefix + hash -> blockData
+	headerPrefix     = []byte("hdr") // headerPrefix + hash -> header
+	blockDataPrefix  = []byte("bld") // blockDataPrefix + hash -> blockData
+	headerHashPrefix = []byte("hsh")
 )
 
+// encodeBlockNumber encodes a block number as big endian uint64
+func encodeBlockNumber(number uint64) []byte {
+	enc := make([]byte, 8)
```

Comment:
Maybe a comment about why we are using `8` here.

Reply:
it's because in the line after we are doing PutUint64, which takes 8 bytes to put the uint64 into